test/benchmark/benchmark_malloc_realloc.cpp

```diff
@@ -0,0 +1,144 @@
+    state.SkipWithError("Size for allocation is too small for this test");
+  }
+  for (auto _ : state) {
+    void * ptr = std::malloc(malloc_size);
+    if (PERFORMANCE_TEST_FIXTURE_UNLIKELY(nullptr == ptr)) {
+      state.SkipWithError("Malloc failed to malloc");
+    }
+    std::free(ptr);
+    benchmark::DoNotOptimize(ptr);
+    benchmark::ClobberMemory();
+  }
+}
+
+BENCHMARK_DEFINE_F(PerformanceTestFixture, benchmark_on_realloc)(
+  benchmark::State & state)
+{
+  const int malloc_size = state.range(1);
+  if (malloc_size < 1) {
+    state.SkipWithError("Size for allocation is too small for this test");
+  }
+  const size_t realloc_size = state.range(2);
+  if (realloc_size < 1) {
+    state.SkipWithError("Size for reallocation is too small for this test");
+  }
+  for (auto _ : state) {
+    void * ptr = std::malloc(malloc_size);
+    if (PERFORMANCE_TEST_FIXTURE_UNLIKELY(nullptr == ptr)) {
+      state.SkipWithError("Malloc failed to malloc");
+    }
+    void * realloc_ptr = std::realloc(ptr, realloc_size);
+    if (PERFORMANCE_TEST_FIXTURE_UNLIKELY(nullptr == realloc_ptr)) {
+      std::free(ptr);
+      state.SkipWithError("Malloc failed to realloc");
+    }
+    std::free(realloc_ptr);
+    benchmark::DoNotOptimize(realloc_ptr);
+    benchmark::ClobberMemory();
+  }
+}
+
+// Malloc sizes Range from 1 to 2^27 each time multiplying by 16. Each value tested
+// with/without performance metrics
+static void malloc_args(benchmark::internal::Benchmark * b)
+{
+  for (int shift_left = 0; shift_left < 32; shift_left += 4) {
+    for (int enable_tracking = kDisablePerformanceTracking;
+      enable_tracking <= kEnablePerformanceTracking; enable_tracking++)
+    {
+      b->Args({enable_tracking, 1 << shift_left});
+    }
+  }
+}
+
+BENCHMARK_REGISTER_F(PerformanceTestFixture, benchmark_on_malloc)
+->ArgNames({"Enable Performance Tracking", "Malloc Size"})->Apply(malloc_args);
+
+// Three types of realloc tests, one where malloc is smaller than realloc, one where they are
+// the same, and one where malloc is larger than realloc. Realloc size ranges from 1 to 2^27
+// each time multiplying by 32. Each stop is tested with/without performance metrics
+static void realloc_args(benchmark::internal::Benchmark * b)
+{
+  for (int malloc_adjustment = -1; malloc_adjustment <= 1; ++malloc_adjustment) {
+    for (int realloc_shift = 0; realloc_shift < 32; realloc_shift += 8) {
+      for (int enable_tracking = kDisablePerformanceTracking;
+        enable_tracking <= kEnablePerformanceTracking; ++enable_tracking)
+      {
+        const int malloc_shift = realloc_shift + malloc_adjustment;
+        if (malloc_shift < 0) {
+          continue;
+        }
+        b->Args({enable_tracking, 1 << malloc_shift, 1 << realloc_shift});
+      }
+    }
+  }
+}
+
+BENCHMARK_REGISTER_F(PerformanceTestFixture, benchmark_on_realloc)
+->ArgNames({"Enable Performance Tracking", "Malloc Size", "Realloc Size"})
+->Apply(realloc_args);
+
+BENCHMARK_MAIN();
```

Comment:
We don't explicitly do this in any other tests. Is there a particular reason it's done here?

```suggestion
```

Reply:
Removed

Reply:
Doesn't appear to have been removed yet.

Reply:
Actually removed